lib/network/http/utils.js

```diff
@@ -0,0 +1,109 @@
+'use strict'; // eslint-disable-line
+
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into an object
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+
+ * @return {Object} object containing a range specification, with
+ * either of:
+ * - start and end attributes: a fully specified range request
+ * - a single start attribute: no end is specified in the range request
+ * - a suffix attribute: suffix range request
+ * - an error attribute of type errors.InvalidArgument if the range
```

Comment:
we should run the ceph tests against s3 server with this branch before merging this. I believe AWS has specific rules on when it ignores an invalid range (just sends the full object) and when it returns an error.